hawkular-inventory-api/src/test/java/org/hawkular/inventory/api/test/AbstractBaseInventoryTestsuite.java

```diff
@@ -2181,6 +2470,105 @@ public void testObserveDataEntities() throws Exception {
+        });
+    }
+
+    @Test
+    public void testObserveIdentityHashChangedOnParentOfCreatedEntity() throws Exception {
+        String tenantId = "testObserveIdentityHashChangedOnParentOfCreatedEntity";
+        testIdentityHashObservation(tenantId, updateCounts -> {
+            //now create some new grandchild of the feed, let's say new sub-resource
+            inventory.tenants().get(tenantId).feeds().get("feed").resources().get("resource").resources()
+                    .create(Resource.Blueprint.builder().withId("childResource").withResourceTypePath
+                            ("../resourceType").build());
+
+            //and check that identity hashes changed and we were informed about it
+            Assert.assertEquals(1, updateCounts.remove(Resource.class).intValue()); //the hash of parent resource
+            Assert.assertEquals(1, updateCounts.remove(Feed.class).intValue());
+            Assert.assertTrue(updateCounts.isEmpty());
+        });
+    }
+
+    @Test
+    public void testObserveIdentityHashChangedOnParentOfDeletedEntity() throws Exception {
+        String tenantId = "testObserveIdentityHashChangedOnParentOfDeletedEntity";
+        testIdentityHashObservation(tenantId, updateCounts -> {
+            //now delete the return type of the operation type
+            inventory.tenants().get(tenantId).feeds().get("feed").resourceTypes().get("resourceType")
+                    .operationTypes().get("operationType").data().delete(returnType);
+
+            //and check that identity hashes changed and we were informed about it
+            Assert.assertEquals(1, updateCounts.remove(OperationType.class).intValue());
+            Assert.assertEquals(1, updateCounts.remove(ResourceType.class).intValue());
+            Assert.assertEquals(1, updateCounts.remove(Feed.class).intValue());
+            Assert.assertTrue(updateCounts.isEmpty());
+        });
+    }
+
+    private void testIdentityHashObservation(String tenantId, Consumer<Map<Class<?>, Integer>> test) throws Exception {
+        Subscription subs = null;
+        try {
+            createInventoryForIdentityHashChangeObservations(tenantId);
+
+            Map<Class<?>, Integer> updateCounts = new HashMap<>();
+
+            subs = Observable.merge(
+                    inventory.observable(Interest.in(Feed.class).having(identityHashChanged())),
+                    inventory.observable(Interest.in(ResourceType.class).having(identityHashChanged())),
+                    inventory.observable(Interest.in(MetricType.class).having(identityHashChanged())),
+                    inventory.observable(Interest.in(OperationType.class).having(identityHashChanged())),
+                    inventory.observable(Interest.in(Resource.class).having(identityHashChanged())),
+                    inventory.observable(Interest.in(Metric.class).having(identityHashChanged())),
+                    inventory.observable(Interest.in(DataEntity.class).having(identityHashChanged()))
```

Comment:
This is the only place where the `.having(identityHashChanged()))` is used, so it's not turned on by default?